### PR TITLE
[finishes #162407549] create a user login page

### PR DESCRIPTION
**Pull request by edkahara**
__What does this PR do?__
Create a user login api endpoint

__Description of tasks to be completed.__
Create an api endpoint that facilitates the logging in of an existing user. Test the api endpoint.

__How should it be manually tested?__
Clone the project. Check out the ft-create-user-login-api-endpoint-#162407549 branch. Start the environment using env\Scripts\activate. Run flask run on the terminal and go to Postman. To perform the tests run nosetest.

__Relevant pivotal tracker stories.__
#162407549

### CODE REVIEW

File: app/api/v1/users/views.py [hunk 1]
```diff
@@ -42,3 +43,27 @@ def post(self):
...
+        data = request.get_json()
+        user = {
+            "username": data["username"],
+            "password": data["password"]
+        }
+        user_to_log_in = self.users.get_specific(user["username"])
+        if user_to_log_in:
+            if user_to_log_in["password"] == user["password"]:
+                access_token = create_access_token(user["username"], expires_delta=datetime.timedelta(days=365))
+                return {"status": 200, "data": [{"id": user_to_log_in["id"], "access_token": access_token, "message": "User Logged In"}]}
```
Comment (bl4ck4ndbr0wn) [inline app/api/v1/users/views.py]: i think 365 days should be token expiry for a developer, not a normal user?

**Review by bl4ck4ndbr0wn — COMMENTED** [review]
Nice.

File: app/api/v1/users/views.py (hunk 1 above)
Comment (edkahara) [reply app/api/v1/users/views.py]: Yea I forgot to remove that while I was testing the API on postman. I'll shorten it.

File: app/__init__.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+from flask_jwt_extended import JWTManager
 
 from .api.v1 import version_one as v1
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object('instance.config')
     app.register_blueprint(v1)
+    jwt = JWTManager(app)
+    app.config['JWT_SECRET_KEY'] = 'edkahara'
```
Comment (thoth-ky) [inline app/__init__.py]: leverage environment variables instead of setting it directly in the code

File: app/api/v1/users/views.py [hunk 3]
```diff
@@ -42,3 +43,27 @@ def post(self):
...
+            else:
+                return {"status": 400, "error": "The password you entered is incorrect. Please try again."}, 400
+        else:
+            return {"status": 404, "error": "The username you entered doesn't belong to an account. Please check your username and try again."}
```
Comment (thoth-ky) [inline app/api/v1/users/views.py]: I think this should be a 401 error

File: app/api/v1/users/views.py [hunk 4]
```diff
@@ -42,3 +43,27 @@ def post(self):
...
+            "username": data["username"],
+            "password": data["password"]
+        }
+        user_to_log_in = self.users.get_specific(user["username"])
+        if user_to_log_in:
+            if user_to_log_in["password"] == user["password"]:
+                access_token = create_access_token(user["username"], expires_delta=datetime.timedelta(days=365))
+                return {"status": 200, "data": [{"id": user_to_log_in["id"], "access_token": access_token, "message": "User Logged In"}]}
+            else:
+                return {"status": 400, "error": "The password you entered is incorrect. Please try again."}, 400
```
Comment (thoth-ky) [inline app/api/v1/users/views.py]: This too should be a 401(Unauthorized)

**Review by thoth-ky — CHANGES_REQUESTED** [review]

File: app/tests/v1/test_users.py [hunk 5]
```diff
@@ -43,3 +47,12 @@ def test_sign_up_unsuccessful_different_passwords(self):
...
         data = json.loads(response.data)
         self.assertEqual(response.status_code, 400)
+
+    def test_log_in_successful(self):
+        response = self.app.post('/api/v1/users/signup', json = self.new_user_same_passwords)
+        data = json.loads(response.data)
+        self.assertEqual(response.status_code, 201)
+
+        response = self.app.post('/api/v1/users/login', json = self.new_user_login_correct_details)
+        data = json.loads(response.data)
```
Comment (thoth-ky) [inline app/tests/v1/test_users.py:68]: Test contents of `response data` not just  `status code`, for all responses

File: instance/config.py [hunk 6]
```diff
@@ -1,3 +1,5 @@
+SECRET_KEY='mysecretkey'
```
Comment (thoth-ky) [inline instance/config.py]: Don't publish the `secret_key`. It should be secret. Leverage `os.getenv()` to get it from the environment

Comment (edkahara) [reply instance/config.py]: So a user who clones the repo should make their own secret key?
Edit: Never mind. I get it now. Let me work on it.

**Pull request merged**